### PR TITLE
Maximize window visual glitch on Windows fix (fix #1447)

### LINKED ISSUE #1447: Maximize window visual glitch on Windows
### Homework

- [X] I took the time to write a good, descriptive issue title
- [X] I read `nativefier --help` and [API.md](https://github.com/nativefier/nativefier/blob/master/API.md).
- [X] I checked [CATALOG.md](https://github.com/nativefier/nativefier/blob/master/CATALOG.md) for community suggestions & workarounds.
- [X] I searched [existing issues, open & closed](https://github.com/nativefier/nativefier/issues?q=is%3Aissue). Yes, my bug is new.
- [X] I'm running the [latest version](https://github.com/nativefier/nativefier/releases).

### Nativefier command

nativefier https://github.com

### Steps to reproduce

1. Create an app with Nativefier
2. Open the app that was created
3. Maximize the window
4. Close the app
5. Open the app again, on Windows a visual glitch can be seen until the content is loaded

### Expected behavior

No visual glitch on Windows.

### Actual behavior

On Windows a visual glitch occurs when opening an app made with Nativefier that had its window maximized.

This is an Electron issue: [[Bug]: Maximize window visual glitch on Windows · Issue #35246 · electron/electron](https://github.com/electron/electron/issues/35246)

Until it's fixed in Electron, a workaround could be used to fix this in Nativefier.

### Debug info

_No response_

### Nativefier version

49.0.0

### Node.js version

18.7.0

### npm version

8.16.0

### OS

Windows 11

### PR DESCRIPTION
**Pull request by Adam777Z**
Fix for [Maximize window visual glitch on Windows · Issue #1447 · nativefier/nativefier](https://github.com/nativefier/nativefier/issues/1447)

### CODE REVIEW

**Comment by ronjouch** [comment]
@Adam777Z hi and thanks for the PR. Can you precise what you mean by "glitch"? A precise description or screenshot would be welcome.

**Comment by Adam777Z** [comment]
This is only a temporary fix until it's fixed in Electron.

There's no change in functionality, it works as before, just without the visual glitch on Windows. This change is for Windows only.

You can test this easily on Windows:
1. Open an app made with the latest version of Nativefier using the default options
2. Maximize the window
3. Close the app
4. Open the app again. Because the window size was saved, it should open maximized, and the visual glitch can be seen without this fix applied.

File: app/src/components/mainWindow.ts [hunk 1]
```diff
@@ -73,7 +73,8 @@ export async function createMainWindow(
     // Whether the window should always stay on top of other windows. Default is false.
     alwaysOnTop: options.alwaysOnTop,
     titleBarStyle: options.titleBarStyle ?? 'default',
-    show: options.tray !== 'start-in-tray',
+    // Maximize window visual glitch on Windows fix
```
Comment (ronjouch) [inline app/src/components/mainWindow.ts:76]:
```suggestion
    // Maximize window visual glitch on Windows fix
    // We want a consistent behavior on all OSes, but Windows needs help to not glitch.
    // So, we manually mainWindow.show() later, see a few lines below
```

File: app/src/components/mainWindow.ts [hunk 2]
```diff
@@ -94,6 +95,9 @@ export async function createMainWindow(
 
   if (options.tray === 'start-in-tray') {
     mainWindow.hide();
+  } else if (process.platform === 'win32') {
+    // Maximize window visual glitch on Windows fix
```
Comment (ronjouch) [inline app/src/components/mainWindow.ts:99]:
```suggestion
    // See other "Maximize window visual glitch on Windows fix" comment above.
```

**Review by ronjouch — APPROVED** [review]

**Pull request merged**